src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
I saw this case when using byte and constants.  Byte as a type is signed, but the indirect store will be signed.

I think that we're guaranteed that we'll get compatible types here, but an assert would be good to add.